### PR TITLE
Logit Softcapping

### PR DESCRIPTION
**Pull request by mmoffatt2**
Added attention and final logit softcapping as implemented [here](https://github.com/google/gemma_pytorch/blob/main/gemma/model.py) for the Gemma 2 model. 

Attention logit softcapping does softcapping right before the masking.

Final logit softcapping does softcapping for the final logits of the model.

### CODE REVIEW

**Comment by gkielian** [comment]
Continuing in PR #468
